frame/assets/src/benchmarking.rs

```diff
@@ -514,8 +514,14 @@ benchmarks_instance_pallet! {
 			SystemOrigin::Signed(new_account.clone()).into(),
 			asset_id
 		).is_ok());
-		// `touch` should reserve some balance of the caller...
-		assert!(!T::Currency::reserved_balance(&new_account).is_zero());
+		// check reserved balance of the caller...
+		if T::AssetAccountDeposit::get() > DepositBalanceOf::<T, I>::zero() {
+			// `touch` should reserve some balance of the caller if `AssetAccountDeposit` is required.
+			assert!(!T::Currency::reserved_balance(&new_account).is_zero());
+		} else {
+			// `touch` should not reserve any balance of the caller if no `AssetAccountDeposit` is required.
+			assert!(T::Currency::reserved_balance(&new_account).is_zero());
+		}
```

Comment:
I suppose using "not zero" works when there are other deposits being reserved, but in this case maybe it's just better to do:
```suggestion
		assert_eq!(T::Currency::reserved_balance(&new_account), T::AssetAccountDeposit::get());
```